kf_lib_data_ingest/etl/stage_analyses.py

```diff
@@ -70,15 +65,57 @@ def check_counts(discovery_sources, expected_counts):
                 },
                 ignore_index=True
             )
-        message.append(
+        logger.info(
+            'EXPECTED COUNT CHECKS\n' +
             tabulate(
                 checks,
                 headers='keys', showindex=False, tablefmt='psql'
             )
         )
 
-    return passed, '\n'.join(message)
+    return passed
 
 
-def compare_counts(discovery_sources_one, discovery_sources_two):
-    False, "Compare Counts NYI"
+def compare_counts(
+    name_one, discovery_sources_one, name_two, discovery_sources_two, logger
+):
+    logger.info(f'COMPARING COUNTS')
+    passed = True
+
+    setA = set(discovery_sources_one.keys())
+    setB = set(discovery_sources_two.keys())
+    diff = setA ^ setB
+    if diff:
+        logger.info(
+            f'❌ Column keys not equal between {name_one} and '
+            f'{name_two}'
+        )
+        logger.debug(
+            f'{name_one} {setA}\n'
+            'vs\n'
+            f'{name_two} {setB}\n'
+            'Difference is:\n' +
+            str(diff)
+        )
+        passed = False
+
+    for k, v in discovery_sources_one.items():
+        setA = set(v.keys())
+        setB = set(discovery_sources_two[k].keys())
+        diff = setA ^ setB
+        if diff:
+            logger.info(
```

Comment:
Seems redundant with line 111

```suggestion
            logger.info(f'❌ Column values for {k} not equal between {name_one} and '
                f'{name_two}  ( # of different values = {len(diff)})'
```
I like keeping the error to a single log statement instead of two separate ones. I think it's easier to search/text process